### PR TITLE
Various core deprecations

### PR DESCRIPTION
**Pull request by Pessimistress**
For #3869 

Mostly unused and undocumented methods.

See inline comments.

### CODE REVIEW

File: modules/core/src/lib/layer.js [hunk 1]
```diff
@@ -586,10 +550,6 @@ export default class Layer extends Component {
     }
     // End subclass lifecycle methods
 
-    // TODO deprecated, for backwards compatibility with older layers
-    // in case layer resets state
-    this.state.attributeManager = this.getAttributeManager();
```
Comment (Pessimistress) [inline modules/core/src/lib/layer.js]: This use case is still supported. This line is a duplicate.

File: modules/core/src/lib/layer.js [hunk 2]
```diff
@@ -713,13 +667,6 @@ export default class Layer extends Component {
     this.props = currentProps;
   }
 
-  // {uniforms = {}, ...opts}
-  pickLayer(opts) {
```
Comment (Pessimistress) [inline modules/core/src/lib/layer.js]: Only `getPickingInfo` is documented.

File: modules/core/src/lib/layer.js [hunk 3]
```diff
@@ -599,12 +559,6 @@ export default class Layer extends Component {
     });
 
     this._updateState();
-
-    const model = this.getSingleModel();
```
Comment (xintongxia) [inline modules/core/src/lib/layer.js]: Why this could be removed?

Comment (Pessimistress) [reply modules/core/src/lib/layer.js]: - Model id does not serve any purpose other than for debugging
- All models created by build-in layers already have ids

**Pull request merged**